crates/bindings/tests/deps.rs

```diff
@@ -2,12 +2,41 @@
 //! to make sure we don't unknowningly add a bunch of dependencies here,
 //! slowing down compilation for every spacetime module.
 
+#[cfg(target_os = "macos")]
+/// On macOS, the `keccak` crate depends on `cpufeatures/libc`, which don't on Linux.
+fn filter_keccak(input: String) -> String {
+    let mut lines = input.lines();
+    let mut found_keccak = false;
+    let mut found_cpufeatures = false;
+    let mut result = Vec::new();
+
+    while let Some(line) = lines.next() {
+        if line.contains("keccak") {
+            found_keccak = true;
+            result.push(line.to_string());
+        } else if found_keccak && line.contains("cpufeatures") {
+            found_cpufeatures = true;
+        } else if found_cpufeatures && line.contains("libc") {
+        } else {
+            result.push(line.to_string());
+        }
+    }
+
+    result.join("\n")
+}
+
+#[cfg(not(target_os = "macos"))]
```

Comment:
Instead of filtering on macos, can we have two different snapshots? I.e. add a new snapshot for macos. And then use a compile-time flag inside of `deptree_snapshot` to use the correct one depending on the platform?